cpp/src/io/orc/reader_impl.cu

```diff
@@ -434,59 +457,114 @@ class aggregate_orc_metadata {
     return selected_stripes_mapping;
   }
 
+  /**
+   * @brief Adds column as per the request and saves metadata about children.
+   *        Struct children are in the same level as struct, only list column
+   *        children are pushed to next level.
+   *
+   * @param selection A vector that saves list of columns as per levels of nesting.
+   * @param types A vector of schema types of columns.
+   * @param level current level of nesting.
+   * @param id current column id that needs to be added.
+   * @param has_timestamp_column True if timestamp column present and false otherwise.
+   *
+   * @return returns number of child columns
```

Comment:
is this just children in the same level?

Reply:
It is same level for struct and next level for list.